backend/spell.py

```diff
@@ -494,35 +476,25 @@ def cast(self,board,str=[]):
         # loop through all rooms, and rearrange the objects in each room
         for room in populated_linked_rooms:
             # move objects that used to be in the room, one at a time.
-            """
-            Why Use Enumeration?: A Short Discussion. 
-            Enumerating locations allows us to pop() locations from the list once
-            they are targeted; python's list.remove() uses "in" to determine membership,
-            which uses equality, which does not behave well for numpy's vectors.
-            A perhaps more Pythonic solution to this would be to define a new location
-            class which carries the data of the location and has the right notion of
-            equivalence implemented. This class could also have getter methods for various
-            coordinate systems. (-JM 20/6/20)
-            """
-            unoccupied_locations = list(enumerate([hex.location for hex in room.hexes]))
+            unoccupied_locations = [hex.location for hex in room.hexes]
             object_location_pairs = []
             for hex in room.hexes:
                 # assign a new location to the object on this hex, if there is one
                 # this assignment goes into object_location_pairs
                 if hex.occupant:
                     # choose a hex not yet targeted 
-                    target_location = choose_from_list(
+                    target_hex_index = screen_input.choose_hexes(
                         board.screen,
-                        unoccupied_locations,
-                        prompt_text="Pick a location for {}".format(hex.occupant)
+                        [location.find_hex(board,loc) for loc in unoccupied_locations],
+                        prompt_text="Pick a hex for {}".format(hex.occupant),
+                        return_index = True
                     )
                     # keep track of the new location for the object
-                    object_location_pairs.append((hex.occupant, target_location[1]))
+                    object_location_pairs.append((hex.occupant, room.hexes[target_hex_index]))
                     hex.occupant = None
-                    unoccupied_locations.pop(target_location[0])
+                    unoccupied_locations.pop(target_hex_index)
             # update the board with new locations
-            for object_to_place, loc in object_location_pairs:
-                target_hex = location.find_hex(board, loc)
+            for object_to_place, target_hex in object_location_pairs:
```

Comment:
with the change I suggested above you'll need to change this back
```suggestion
            for object_to_place, loc in object_location_pairs:
                target_hex = location.find_hex(board, loc)
```